stl/inc/xmemory

```diff
@@ -677,35 +741,76 @@ struct _Default_allocator_traits { // traits for std::allocator
     template <class _Other>
     using rebind_traits = allocator_traits<allocator<_Other>>;
 
-    _NODISCARD static __declspec(allocator) pointer allocate(_Alloc&, _CRT_GUARDOVERFLOW const size_type _Count) {
-        return static_cast<pointer>(_Allocate<_New_alignof<value_type>>(_Get_size_of_n<sizeof(value_type)>(_Count)));
+    _NODISCARD static _CONSTEXPR20_DYNALLOC __declspec(allocator) pointer
+        allocate(_Alloc& _Al, _CRT_GUARDOVERFLOW const size_type _Count) {
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
+            return _Al.allocate(_Count);
+        } else
+#endif // __cpp_lib_is_constant_evaluated

```

Comment:
I believe I was the culprit here.

From my understanding there is a difference between MSCV and clang.

MSVC will replace `_Al.allocate` with fairy dust, as this is `allocator::allocate`
clang will go into `_Allocate`

Reply:
This comment was intended to point out that we have escape hatches for `constexpr` things in both this function and in the `_Allocate` function that we are evidently avoiding if `std::allocator::allocate()` is called in a `constexpr` context.  It seems like it would be nice to centralize the logic for workarounds for `constexpr` things in one area and sort of determine whether `_Allocate` is completely safe for `constexpr` or not.  This is definitely not something that needs to be included in this PR, but I want to file an issue for it to at least take another look to see if we can make this dance a little nicer given the approaches from Clang and MSVC.  

From my understanding: 
MSVC does fairy magic when it sees `std::allocator::allocate`, but doesn't otherwise allow `::operator new` so we need to go through `std::allocator::allocate` in `allocator_traits::allocate`
Clang allows calls to `::operator new` if they are reached transitively from an evaluation of `std::allocator::allocate` so we can maintain the call to `_Allocate` directly, or go through `std::allocator::allocate` which does a call to `_Allocate`.

Reply:
Follow up here: I've filed GH-1532 to track this so the transition comment can read `// TRANSITION, GH-1532`.